moveit_ros/planning/plan_execution/src/plan_execution.cpp

```diff
@@ -517,7 +517,7 @@ void plan_execution::PlanExecution::successfulTrajectorySegmentExecution(const E
   if (index < plan->plan_components_.size() && plan->plan_components_[index].trajectory_ &&
       !plan->plan_components_[index].trajectory_->empty())
   {
-    if (!isRemainingPathValid(*plan, std::make_pair<int>(index, 0)))
+    if (!isRemainingPathValid(*plan, std::make_pair<int>(static_cast<int>(index), 0)))
```

Comment:
@seanyen, what was the actual problem here?
For me, `std::make_pair<int>` doesn't make sense in the first place. There should be two template arguments provided,
i.e. `std::make_pair<int, int>`. Would this solve your initial issue?

Reply:
usually, `std::make_pair(index,0)` should be fine here.
I would guess the first template parameter was provided to enforce a cast of the size_t - which should have been done explicitly in the first place.
That would mean the current patch can simply drop the template type too.

Alternatively, it could also provide both types (which probably reads better to some people), but this will likely not resolve the problem.

Reply:
> Usually, `std::make_pair(index,0)` should be fine here.

This would throw a warning because the resulting type would be `std::pair<ssize_t, int>`.

> Alternatively, it could also provide both types, i.e. `std::make_pair<int,int>(index,0)`, but this will likely not resolve the problem.

Indeed, I would prefer this over `std::make_pair<static_cast<int>(index),0)`. However, I'm not sure about the actual problem in Windows. Maybe `std::make_pair<int>()` is just not available as I assume...

Reply:
@seanyen, please could you report on the actual error you got on Windows here, so that we can decide which solution to prefer?

Reply:
Thanks. Looks like Windows denies to "downcast" to lower bit resolution. In this case we cannot avoid the explicit cast. Thus, I suggest:
`std::make_pair<static_cast<int>(index),0)`

Reply:
I thought about this too, but the problem is that [TrajectoryExecutionManager::getCurrentExpectedTrajectoryIndex](https://github.com/ros-planning/moveit/blob/master/moveit_ros/planning/plan_execution/src/plan_execution.cpp#L269)
is meant to be compatible with this function and it's return type is `std::pair<int,int>`, because it needs to be able to express `-1` for "no segment" and "did not start (or do not monitor) execution of part yet".
I would have argued for `ssize_t`, but that's not cross-platform.

So if we would go for the `size_t` approach (in theory for *both* parameters) we would need to revise the semantics and API of the execution manager too. Last time I looked at it I rejected this as unnecessarily intrusive (and after all, try to create a motion plan with more than `INT_MAX` components or waypoints...).

Reply:
@v4hn Thanks for the context. I reverted the change back to `static_cast<int>` approach.